src/main/java/com/fasterxml/jackson/databind/introspect/POJOPropertiesCollector.java

```diff
@@ -1128,8 +1128,13 @@ protected void _renameUsing(Map<String, POJOPropertyBuilder> propMap,
         for (POJOPropertyBuilder prop : props) {
             PropertyName fullName = prop.getFullName();
             String rename = null;
-            // [databind#4302] since 2.17, Need to skip renaming for Enum properties
-            if (!prop.hasSetter() && prop.getPrimaryType().isEnumType()) {
+            // [databind#4302] since 2.16, Need to skip renaming for Enum instances (but not for Enum fields)
+            if (!prop.hasSetter()
+                    && Objects.nonNull(prop.getField())
+                    && (Objects.equals(_type, prop.getField().getType()) && prop.getPrimaryType().isEnumType())
+            ) {
```

Comment:
I wonder if there is cleaner solution for this... 🤔 
Or just extract method to some where (probably not `POJOPropertyBuilder`, it doens't need to know about Enum stuff)

Reply:
Sorry, I should have noticed -- "fix" is actually totally wrong.

It applies to properties with Enum value but what we want are properties WITHIN type that has properties.

I am not sure how it would have resolved the problem but I can look into fixing the problem. Odd databind had no test for naming strategy with Enum-valued POJO property.

Reply:
First thing I think (beyond revert) is to add a test for naming strategy. I can do that.